src/Algebra/Morphism/Construct/Composition.agda

```diff
@@ -0,0 +1,404 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- The composition of morphisms between algebraic structures.
+------------------------------------------------------------------------
+
+{-# OPTIONS --safe --without-K #-}
+
+module Algebra.Morphism.Construct.Composition where
+
+open import Algebra.Bundles
+open import Algebra.Morphism.Structures
+open import Data.Product
+open import Function.Base using (_∘_)
+open import Level using (Level)
+open import Relation.Binary.Morphism.Construct.Composition
+
+private
+  variable
+    a b c ℓ₁ ℓ₂ ℓ₃ : Level
+
+module _
+  {M₁ : Magma a ℓ₁} {M₂ : Magma b ℓ₂} {M₃ : Magma c ℓ₃}
+  {f : Magma.Carrier M₁ → Magma.Carrier M₂}
+  {g : Magma.Carrier M₂ → Magma.Carrier M₃}
+  where
+
+  open Magma
+  private
+    module M₁ = Magma M₁
+    module M₂ = Magma M₂
+    module M₃ = Magma M₃
+
+  isMagmaHomomorphism
+    : IsMagmaHomomorphism M₁.rawMagma M₂.rawMagma f
+    → IsMagmaHomomorphism M₂.rawMagma M₃.rawMagma g
+    → IsMagmaHomomorphism M₁.rawMagma M₃.rawMagma (g ∘ f)
+  isMagmaHomomorphism f-homo g-homo = record
+    { isRelHomomorphism = isRelHomomorphism f-homo.isRelHomomorphism g-homo.isRelHomomorphism
+    ; homo = λ x y → M₃.trans (g-homo.⟦⟧-cong (f-homo.homo x y)) (g-homo.homo (f x) (f y))
+    }
+    where
+      module f-homo = IsMagmaHomomorphism f-homo
```

Comment:
You could shorten these modules just to `F` and `G` which would cut out a lot of noise in the record construction.